cmd/kubeops/main.go

```diff
@@ -201,12 +201,13 @@ func parseClusterConfig(configPath, caFilesPrefix string) (kube.ClustersConfig,
 	configs := kube.ClustersConfig{Clusters: map[string]kube.ClusterConfig{}}
 	configs.PinnipedProxyURL = pinnipedProxyURL
 	for _, c := range clusterConfigs {
-		if c.APIServiceURL == "" {
-			if configs.KubeappsClusterName == "" {
-				configs.KubeappsClusterName = c.Name
-			} else {
-				return kube.ClustersConfig{}, nil, fmt.Errorf("only one cluster can be configured without an apiServiceURL, two defined: %q, %q", configs.KubeappsClusterName, c.Name)
-			}
+		// Select the cluster in which Kubeapps in installed. We look for either
+		// `isKubeappsCluster: true` or an empty `APIServiceURL`.
+		isKubeappsClusterCandidate := c.IsKubeappsCluster || c.APIServiceURL == ""
+		if isKubeappsClusterCandidate && configs.KubeappsClusterName == "" {
+			configs.KubeappsClusterName = c.Name
+		} else if isKubeappsClusterCandidate && configs.KubeappsClusterName != "" {
+			return kube.ClustersConfig{}, nil, fmt.Errorf("only one cluster can be configured using either 'isKubeappsCluster: true' or without an apiServiceURL to to refer to the cluster on which Kubeapps is installed, two defined: %q, %q", configs.KubeappsClusterName, c.Name)
```

Comment:
Might be slightly simpler to parse (mentally) as:

```suggestion
		if isKubeappsClusterCandidate {
			if configs.KubeappsClusterName == "" {
				configs.KubeappsClusterName = c.Name
			} else {
				return kube.ClustersConfig{}, nil, fmt.Errorf("only one cluster can be configured using either 'isKubeappsCluster: true' or without an apiServiceURL to to refer to the cluster on which Kubeapps is installed, two defined: %q, %q", configs.KubeappsClusterName, c.Name)
			}
```

but feel free to ignore.